compiler/front/cli_reporter.nim

```diff
@@ -4502,6 +4502,23 @@ func astDiagToLegacyReport(conf: ConfigRef, diag: PAstDiag): Report {.inline.} =
   result =
     case kind
     of repVMKinds:
+      if diag.kind == adVmGenError and diag.duringJit or
+          diag.kind == adVmError:
+        let st =
+          case diag.kind
+          of adVmGenError: diag.vmGenTrace
+          of adVmError:    diag.vmTrace
+          else:            unreachable()
+
+        vmRep.trace =
+            (ref VMReport)(kind: rvmStackTrace,
+                      currentExceptionA: st.currentExceptionA,
+                      currentExceptionB: st.currentExceptionB,
+                      stacktrace: st.stacktrace,
+                      skipped: st.skipped,
+                      location: some st.location,
+                      reportInst: toReportLineInfo(st.instLoc))
```

Comment:
```suggestion
        {.cast(noSideEffect).}: # side-effect analysis fails
          vmRep.trace =
            (ref VMReport)(kind: rvmStackTrace,
                      currentExceptionA: st.currentExceptionA,
                      currentExceptionB: st.currentExceptionB,
                      stacktrace: st.stacktrace,
                      skipped: st.skipped,
                      location: some st.location,
                      reportInst: toReportLineInfo(st.instLoc))
```
The side-effect analysis is failing here. While I haven't tested the above, it *should* work.